src/main/java/core/basesyntax/DataConverterImpl.java

```diff
@@ -0,0 +1,40 @@
+import java.util.ArrayList;
+import java.util.List;
+
+public class DataConverterImpl implements DataConverter {
+    @Override
+    public List<FruitTransaction> convertToTransaction(List<String> data) {
+        List<FruitTransaction> transactions = new ArrayList<>();
+        for (String line : data) {
+            String[] parts = line.split(",");
+            if (parts.length != 3) {
+                throw new IllegalArgumentException("Invalid data format: " + line);
+            }
+
+            FruitTransaction.Operation operation = getOperation(parts[0]);
```

Comment:
Using a method like 'getOperation' which iterates through all operations every time can be inefficient. Consider using a map to improve the performance.